src/content/docs/extensions/llm.mdx

```diff
@@ -0,0 +1,232 @@
+```
+
+See the list of providers and their embedding documentation.
+
+* [Open-AI](https://platform.openai.com/docs/guides/embeddings)
+* [Amazon Bedrock](https://docs.aws.amazon.com/bedrock/latest/userguide/titan-embedding-models.html)
+* [Google Vertex](https://cloud.google.com/vertex-ai/generative-ai/docs/embeddings/get-text-embeddings)
+* [Google Gemini](https://ai.google.dev/gemini-api/docs/embeddings)
+* [Voyage AI](https://docs.voyageai.com/docs/embeddings)
+* [Ollama](https://ollama.com/blog/embedding-models)
+
+:::note[Note]
+Model parameters are dynamically validated at runtime. If an unsupported model is specified, the provider's API will return an appropriate error.
+:::
+---
+
+## Usage
+
+### Configuration
+
+The LLM extension supports configurable options such as credentials, dimensions, and region where applicable.
+
+- All credentials are provided through environment variables.
+- Dimensions and region are passed to the `CREATE_EMBEDDING` function call, when supported.
+
+The following environment variables must be set for the corresponding providers:
+
+| Provider | Required Environment Variables |
+| --- | --- |
+| `amazon-bedrock` | `AWS_ACCESS_KEY`, `AWS_SECRET_ACCESS_KEY` |
+| `google-vertex` | `GOOGLE_CLOUD_PROJECT_ID`, `GOOGLE_VERTEX_ACCESS_KEY` |
+| `google-gemini` | `GOOGLE_GEMINI_API_KEY` |
+| `open-ai` | `OPENAI_API_KEY` |
+| `voyage-ai` | `VOYAGE_API_KEY` |
+
+:::note[Note]
+The `ollama` provider requires no environment variables but does require the embedding model to be available at `http://localhost:11434`.
+:::
+---
+
+## Embedding Function
+
+The core feature of the LLM extension is a scalar function that generates text embeddings.
+
+### Syntax
+
+```sql
+CREATE_EMBEDDING(prompt: STRING, provider: STRING, model: STRING) -> LIST<FLOAT>
+```
+
+### Overloads
+
+```sql
+CREATE_EMBEDDING(prompt: STRING, provider: STRING, model: STRING, dimensions: INT64) -> LIST<FLOAT>
+```
+
+```sql
+CREATE_EMBEDDING(prompt: STRING, provider: STRING, model: STRING, region: STRING) -> LIST<FLOAT>
+```
+
+```sql
+CREATE_EMBEDDING(prompt: STRING, provider: STRING, model: STRING, dimensions: INT64, region: STRING) -> LIST<FLOAT>
+```
+
+### Parameters
+
+- **prompt**: The input text to embed.
+- **provider**: The embedding provider, e.g., `'open-ai'`, `'ollama'`, `'voyage-ai'`, etc.
+- **model**: The identifier of the embedding model. This may change overtime; reference the provider's API reference before use.
+- **dimensions** *(optional)*: The size of the embedding vector to be returned, if supported by the provider. Must be a positive integer.
+- **region** *(optional)*: A region or endpoint hint, if supported by the provider.
+
+### Examples
+
+```sql
+CREATE_EMBEDDING("Hello world", "openai", "text-embedding-3-small")
+```
+
+```sql
+CREATE_EMBEDDING("Hello world", "ollama", "nomic-embed-text")
+```
+
+```sql
+CREATE_EMBEDDING("Hello world", "google-gemini", "gemini-embedding-exp-03-07")
+```
+
+With optional parameters:
```

Comment:
Try and see if looks better when all these are merged within the same `sql` block separated by newlines.

Currently there are too many blocks.

Also, a new standard I've been thinking we should adopt: try to wrap the statements such that there is no horizontal scrollbar.